front/app/components/admin/ContentBuilder/Widgets/FileAttachment/index.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+
+import {
+  Box,
+  colors,
+  Select,
+  Spinner,
+  Text,
+} from '@citizenlab/cl2-component-library';
+import { useNode } from '@craftjs/core';
+import { useParams } from 'react-router-dom';
+
+import useFileById from 'api/files/useFileById';
+import useFiles from 'api/files/useFiles';
+
+import ButtonWithLink from 'components/UI/ButtonWithLink';
+import FileDisplay from 'components/UI/FileAttachments/FileDisplay';
+
+import { useIntl } from 'utils/cl-intl';
+
+import messages from './messages';
+
+type FileAttachmentProps = {
+  fileId?: string;
+};
+
+const FileAttachment = ({ fileId }: FileAttachmentProps) => {
+  const { data: file } = useFileById(fileId);
+
+  return (
+    <Box id="e2e-file-attachment" maxWidth="1200px" margin="0 auto">
+      <FileDisplay fileNewType={file?.data} />
+    </Box>
+  );
+};
+
+const FileAttachmentSettings = () => {
+  const {
+    actions: { setProp },
+    fileId,
+  } = useNode((node) => ({
+    fileId: node.data.props.fileId,
+    id: node.id,
+  }));
+
+  const { formatMessage } = useIntl();
+
+  const { projectId } = useParams();
+
+  // Get files for project
+  const { data: files, isFetching: isFetchingFiles } = useFiles({
+    project: [projectId || ''],
```

Comment:
```suggestion
    project: projectId ? [projectId] : [],
```

I know this probably will never happen, but an array with one empty string does not seem like a logical fallback value to indicate 'no projects' to me. Maybe something like this?

Reply:
Instead of an empty array, you could also put `undefined` I guess

Reply:
Good point - thanks!!